src/browser/AxePage.ts

```diff
@@ -86,82 +92,20 @@ export function getRunOptions(
- * https://github.com/dequelabs/axe-core/issues/1041).
+ * Used to prevent concurrent runs of `axe.run`, which breaks (see https://github.com/dequelabs/axe-core/issues/1041).
+ * This should never happen, but in the past errors during rendering at the right/wrong time has
+ * caused the next test to start before the previous has stopped.
  *
- * We avoid that by forcing one `axe.run` call to finish before the next one can start. Got the
- * idea from https://github.com/dequelabs/agnostic-axe/pull/6.
+ * Got the idea from https://github.com/dequelabs/agnostic-axe/pull/6.
  */
 function addPromiseQueue() {
-  type QueuedPromise<T> = {
-    promiseCreator: () => Promise<T>;
-    resolve: (value: T | PromiseLike<T>) => void;
-    reject: (reason?: unknown) => void;
-  };
-
-  /**
-   * Queue of promises, which forces any promises added to it to run one at a time.
-   *
-   * This was originally implemented as an ES6 class. But I ran into a lot of issues with how the
-   * class's properties were compiled not working in different environments and browsers, and even
-   * breaking when Babel was updated.
-   *
-   * To avoid that, I've instead implemented this as a function that maintains some state within a
-   * closure. Since there are no class properties (which can be compiled by Babel in different ways),
-   * there are no more problems.
-   */
-  function PromiseQueue<T>() {
-    const pending: QueuedPromise<T>[] = [];
-    let working = false;
-
-    /**
-     * Add a promise to the queue. Returns a promise that is resolved or rejected when the added
-     * promise eventually resolves or rejects.
-     */
-    function add(promiseCreator: () => Promise<T>): Promise<T> {
-      return new Promise((resolve, reject) => {
-        pending.push({promiseCreator, resolve, reject});
-        dequeue();
-      });
-    }
-
-    /**
-     * Run the next promise in the queue.
-     */
-    function dequeue() {
-      // If we're already working on a promise, do nothing.
-      if (working) {
-        return;
-      }
-
-      const nextPromise = pending.shift();
-
-      // If there are no promises to work on, do nothing.
-      if (!nextPromise) {
-        return;
-      }
+  let queue = Promise.resolve();
 
-      working = true;
-
-      // Execute the promise. When it's done, start working on the next.
-      nextPromise
-        .promiseCreator()
-        .then(nextPromise.resolve, nextPromise.reject)
-        .finally(() => {
-          working = false;
-          dequeue();
-        });
-    }
-
-    return {add};
-  }
-
-  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-  // @ts-ignore This function executes in a browser context.
-  window.axeQueue = PromiseQueue<AxeResults>();
+  window.enqueuePromise = function <T>(createPromise: () => Promise<T>) {
+    return new Promise<T>((resolve, reject) => {
+      queue = queue.then(createPromise).then(resolve, reject);
+    });
+  };
```

Comment:
The previous implementation would keep running even if any of the queued promises is rejected (I think).

This new one will stop if any are rejected.

I think that's fine, because:
- It's much simpler
- Looks like the `axe.run` promises are only rejected if there's an actual problem creating the report, **_not_** when there are a11y violations. See https://github.com/dequelabs/axe-core/blob/154ff6d0cb0308e61b99744e5a2df01dbfb768b3/lib/core/public/run.js#L63-L67

In any case, the integration tests in this project are good, and I've tested it as many ways as I could think of.

Reply:
Actually, maybe it won't stop when a promise is rejected, because the rejection is being handled? 🤔